source/version_generated/BUILD

```diff
@@ -8,6 +8,6 @@ envoy_package()
 
 envoy_cc_library(
     name = "version_generated",
-    srcs = ["//:version_generated.cc"],
+    srcs = ["@git_version//:version_generated.cc"],
```

Comment:
We should basically nuke this file once we have the `cc_library` change below, since it won't do anything. Then, anyone who wants `//source/version_generated` as a `dep` should get an `external_dep` to `version_generated`. We should `bind` `version_generated` to `@git_version//:version_generated` in the `def git_version():` to make that work. Then.. we can easily handle this during our import :)